storage: tidy microbenchmarks

jbowens opened the pull request:
Tidy up the pkg/storage microbenchmarks.

jbowens committed:
storage: remove _Pebble suffix from benchmark names

These suffixes are a vestige from the migration from RocksDB to Pebble. All
Engines use Pebble now, and the extra suffix is superfluous.

Epic: none
Release note: none

jbowens committed:
storage: consolidate microbenchmarks

Move microbenchmarks from bench_pebble_test.go into bench_test.go.

Epic: none
Release note: none

jbowens committed:
storage: update BenchmarkBatchBuilderPut

This benchmark was only exercising pebble code. This updates the benchmark to
measure PutMVCC on a pkg/storage Batch.

Epic: none
Release note: None

jbowens committed:
storage: don't use require.* helpers in benchmarked code

Avoid use of the require package helpers in measured benchmark code. The use of
Helper() can confound results, especially in benchmarks of short, fast
sections.

Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/133275)
<!-- Reviewable:end -->

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>  Nice!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/133275#-:-O9uTSBT2EYwLNJaBxXD:b-bsmgd)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

jbowens committed:
storage: use batches in microbenchmarks

Many of the MVCC microbenchmarks tested using the Engine as the ReadWriter or
Writer implementation rather than a batch. Operations performed against the
Engine have very different performance characteristics because they never need
to build a batch skiplist, they don't have the batch's buffers available for
intermediary results, resulting in MVCCEncode allocations, etc. This commit
updates many of these microbenchmarks to create, use and commit a batch
instead. It also removes a MVCCPut benchmark that is made obsolete by a
different microbenchmark that does use a batch.

Epic: none
Release note: none

jbowens commented:
TFTR!

bors r=RaduBerinde
